paimon-core/src/main/java/org/apache/paimon/table/source/DataTableStreamScan.java

```diff
@@ -198,6 +202,19 @@ private Plan nextPlan() {
         }
     }
 
+    private boolean checkDelaySnapshot(long snapshotId) {
+        if (options.scanDelayDuration() == null) {
```

Comment:
extract a class field: `Long scanDelayMillis`.

Reply:
Done, had extract to a class filed.